airflow/www/static/js/api/useSetDagRunNotes.ts

```diff
@@ -0,0 +1,61 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import axios from 'axios';
+import { useMutation, useQueryClient } from 'react-query';
+import { getMetaValue } from 'src/utils';
+import useErrorToast from '../utils/useErrorToast';
+import type { GridData } from './useGridData';
+import { emptyGridData } from './useGridData';
+
+const setDagRunNotesURI = getMetaValue('set_dag_run_note');
+
+export default function useSetDagRunNotes(
```

Comment:
This worked for SetDagRunNote, but it doesn't work for items that require a `body` because there seems to be a bug that marks these fields as optional even though I put that the `body` is required.

```
      requestBody:
        description: Parameters of set Task Instance note.
        required: true
        content:
          application/json:
            schema:
              $ref: '#/components/schemas/SetTaskInstanceNote'
```
Leads to
```
    SetTaskInstanceNote: {
      /** @description The map_index of the TaskInstance. */
      map_index?: number;
      /** @description The custom note to set for this Task Instance. */
      notes?: string;
    };
```

Reply:
By default all object properties are optional, you need to mark them as `required` if needed, for instance here, in the schema, add the 'required' field.
```yaml
    SetTaskInstanceNote:
      type: object
      required:
        - map_index
      properties:
        map_index:
          description: The map_index of the TaskInstance.
          type: integer
        notes:
          description: The custom note to set for this Task Instance.
          type: string
```

will produce:
```ts
    SetTaskInstanceNote: {
      /** @description The map_index of the TaskInstance. */
      map_index: number;
      /** @description The custom note to set for this Task Instance. */
      notes?: string;
    };

```

Reply:
I wasn't aware of this. Thanks!!